echopop/graphics/plotting.py

```diff
@@ -0,0 +1,769 @@
+    ax.yaxis.set_major_locator(y_locator)
+
+    # Return plotting parameter/data dictionary
+    return dict(
+        extent=extent,
+        length_labels=length_labels,
+        age_labels=age_labels,
+        heatmap_array=heatmap_array,
+        delta_age=delta_age,
+        delta_length=delta_length,
+    )
+
+
+def spatial_mesh(
+    x: np.typing.NDArray[float],
+    y: np.typing.NDArray[float],
+    z: np.typing.NDArray[float],
+    geo_config: Dict[str, Any],
+):
+
+    # Get the geo_config settings for transformation
+    # ---- Starting
+    init = geo_config.get("init")
+    # ---- Create a projection object
+    projection_init = pyproj.Proj(init)
+    # ---- Transform the coordinates
+    coords_init = projection_init(x, y)
+
+    # Swap to mercator
+    projection = pyproj.Proj(proj="merc", lat_ts=coords_init[1].mean())
+    # ---- Retransform
+    coords = projection(coords_init[0], coords_init[1])
+
+    # Define processing chain
+    # ---- Spacing
+    spacing = 2.5 / 60
+    # ---- Chain
+    chain = vd.Chain(
+        [
+            ("reduce", vd.BlockReduce("mean", spacing * 111e3)),
+            ("spline", vd.KNeighbors(k=5, reduction=np.mean)),
+        ]
+    )
+    # ---- Fit the chain
+    chain.fit(coords, z)
+
+    # Create grid object
+    # ---- Define region
+    region = vd.get_region((x, y))
+    # ---- Create grid
+    coord_grid = chain.grid(
+        region=region,
+        spacing=spacing,
+        projection=projection,
+    )
+    # ---- Mask the grid
+    coord_mask = vd.distance_mask(
+        data_coordinates=(x, y),
+        maxdist=spacing * 111e3,
+        grid=coord_grid,
+        projection=projection,
+    )
+
+    # Return the masked coordinate values
+    return coord_mask["scalars"]
+
+
+def add_heatmap_grid(
+    ax: Axes,
+    age_labels: np.ndarray,
+    delta_age: float,
+    delta_length: float,
+    length_labels: np.ndarray,
+    **kwargs,
+):
+
+    # Create linear offsets for grid centering
+    # ---- Age
+    age_offset = delta_age / 2
+    # ---- Length
+    length_offset = delta_length / 2
+
+    # Create list increments for each axis
+    # ---- Age
+    age_increments = (age_labels - age_offset).tolist()
+    # ---- Length
+    length_increments = (length_labels - length_offset).tolist()
+
+    # Add to the plot
+    # ---- Vertical lines (age, x)
+    ax.vlines(
+        age_increments,
+        ymin=length_labels.min() - length_offset,
+        ymax=length_labels.max() + length_offset,
+        colors="black",
+    )
+    # ---- Horizontal lines (length, y)
+    ax.hlines(
+        length_increments,
+        xmin=age_labels.min() - age_offset,
+        xmax=age_labels.max() + age_offset,
+        colors="black",
+    )
+
+
+def plot_age_length_distribution(
+    data_dict: Dict[str, Any],
+    **kwargs,
+):
+
+    # Get the dataset variable name
+    variable = kwargs.get("variable", None)
+
+    # Get sex
+    sex = kwargs.get("sex", None)
```

Comment:
Why do you need these if the inputs are already validated? If either of the variables are `None`, the subsequence code would error out also.